<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="beaglebone">
	<xacro:property name="bodyWidth" value="0.36" />
	<xacro:property name="bodyLength" value="0.44" />
	<xacro:property name="bodyHeigth" value="0.054" />
	<xacro:property name="wheelRadius" value="0.0825" />
	<xacro:property name="wheelWidth" value="0.092" />
	<xacro:property name="M_PI" value="3.1415926535" />
	<link name="base_footprint">
		<visual>
			<geometry>
				<box size="0.001 0.001 0.001" />
			</geometry>
		</visual>
	</link>

	<link name="base_link">
		<visual>
			<geometry>
				<box size="${bodyWidth} ${bodyLength} ${bodyHeigth}"/>
			</geometry>
			<material name="gray">
				<color rgba="0.5 0.5 0.5 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.2" radius="0.4" />
			</geometry>
		</collision>
	</link>

	<joint name="base_footprint_joint" type="fixed">
		<parent link="base_footprint" />
		<child link="base_link" />
		<origin rpy="0 0 ${-0.5*M_PI}"/>
	</joint>
	
	<link name="wheelFrontLeft">
		<visual>
			<geometry>
				<cylinder length="${wheelWidth}" radius="${wheelRadius}" />
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1" />
			</material>
			<origin rpy="0 1.57075 0" xyz="-${bodyHeigth} 0 0" />
		</visual>
	</link>
	
	<link name="wheelFrontRight">
		<visual>
			<geometry>
				<cylinder length="${wheelWidth}" radius="${wheelRadius}" />
			</geometry>
			<material name="black"/>
			<origin rpy="0 1.57075 0" xyz="${bodyHeigth} 0 0" />
		</visual>
	</link>
	
	<link name="wheelRearLeft">
		<visual>
			<geometry>
				<cylinder length="${wheelWidth}" radius="${wheelRadius}" />
			</geometry>
			<material name="black"/>
			<origin rpy="0 1.57075 0" xyz="-${bodyHeigth} 0 0" />
		</visual>
	</link>
	
	<link name="wheelRearRight">
		<visual>
			<geometry>
				<cylinder length="${wheelWidth}" radius="${wheelRadius}" />
			</geometry>
			<material name="black"/>
			<origin rpy="0 1.57075 0" xyz="${bodyHeigth} 0 0" />
		</visual>
	</link>

	<link name="laser_scanner">
		<visual>
			<origin rpy="0 0 0" xyz="0 0 ${wheelRadius}"/>
			<geometry>
				<cylinder radius="0.03" length="0.01" />
			</geometry>
			<material name="white">
				<color rgba="1 1 1 1"/>
			</material>
		</visual>
	</link>
	
	<link name="base_imu">
		<visual>
			<geometry>
				<box size="0.001 0.001 0.001" />
			</geometry>
			<material name="white">
				<color rgba="1 1 1 1"/>
			</material>
		</visual>
	</link>
	
	<joint name="base_to_front_left_wheel" type="fixed">
		<parent link="base_link"/>
		<child link="wheelFrontLeft"/>
		<origin xyz="-0.18 0.19 0"/>
	</joint>
	
	<joint name="base_to_front_right_wheel" type="fixed">
		<parent link="base_link"/>
		<child link="wheelFrontRight"/>
		<origin xyz="0.18 0.19 0"/>
	</joint>
	
	<joint name="base_to_rear_left_wheel" type="fixed">
		<parent link="base_link"/>
		<child link="wheelRearLeft"/>
		<origin xyz="-0.18 -0.19 0"/>
	</joint>		

	
	<joint name="base_to_rear_right_wheel" type="fixed">
		<parent link="base_link"/>
		<child link="wheelRearRight"/>
		<origin xyz="0.18 -0.19 0"/>
	</joint>

	<joint name="base_to_laser" type="fixed">
		<parent link="base_link"/>
		<child link="laser_scanner"/>
		<origin xyz="0 0.18 0" rpy="0 0 ${-0.5*M_PI}"/>
	</joint>
	
	<joint name="base_to_imu" type="fixed">
		<parent link="base_link"/>
		<child link="base_imu"/>
		<origin xyz="0 0 0.001" rpy="0 0 0"/>
	</joint>
</robot>
